From f6242c4ffe14b35d5fe7ce469586a67a4b92470e Mon Sep 17 00:00:00 2001
From: Chen Qi <Qi.Chen@windriver.com>
Date: Thu, 20 Sep 2018 14:47:20 +0800
Subject: [PATCH 2/2] fix to build with boost 1.68

Upstream will not compile such an old version with boost 1.68, so mark
this patch as 'WRLinux Specific'.

Upstream-Status: Inappropriate [WRLinux Specific]

Signed-off-by: Chen Qi <Qi.Chen@windriver.com>
---
 src/rbd_replay/Replayer.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/rbd_replay/Replayer.cc b/src/rbd_replay/Replayer.cc
index b37f226..4a5a385 100644
--- a/src/rbd_replay/Replayer.cc
+++ b/src/rbd_replay/Replayer.cc
@@ -336,7 +336,7 @@ void Replayer::wait_for_actions(const action::Dependencies &deps) {
     dout(DEPGRAPH_LEVEL) << "Finished waiting for " << dep.id << " after " << micros << " microseconds" << dendl;
     // Apparently the nanoseconds constructor is optional:
     // http://www.boost.org/doc/libs/1_46_0/doc/html/date_time/details.html#compile_options
-    boost::system_time sub_release_time(action_completed_time + boost::posix_time::microseconds(dep.time_delta * m_latency_multiplier / 1000));
+    boost::system_time sub_release_time(action_completed_time + boost::posix_time::microseconds(long(dep.time_delta * m_latency_multiplier / 1000)));
     if (sub_release_time > release_time) {
       release_time = sub_release_time;
     }
-- 
2.7.4

